packages/@expo/metro-config/src/rewriteRequestUrl.ts

```diff
@@ -1,32 +1,70 @@
 // Copyright 2023-present 650 Industries (Expo). All rights reserved.
-import { getPackageJson } from '@expo/config';
+import { ExpoConfig, getConfig } from '@expo/config';
 import { resolveEntryPoint } from '@expo/config/paths';
 import chalk from 'chalk';
+import fs from 'fs';
 import path from 'path';
 
 import { getServerRoot } from './getModulesPaths';
 
 const debug = require('debug')('expo:metro:config:rewriteRequestUrl');
 
+function directoryExistsSync(file: string): boolean {
+  try {
+    return fs.statSync(file)?.isDirectory() ?? false;
+  } catch {
+    return false;
+  }
+}
+
+function isEnableHermesManaged(
+  expoConfig: Partial<Pick<ExpoConfig, 'ios' | 'android' | 'jsEngine'>>,
+  platform: string
+): boolean {
+  switch (platform) {
+    case 'android': {
+      return (expoConfig.android?.jsEngine ?? expoConfig.jsEngine) !== 'jsc';
+    }
+    case 'ios': {
+      return (expoConfig.ios?.jsEngine ?? expoConfig.jsEngine) !== 'jsc';
+    }
+    default:
+      return false;
+  }
+}
```

Comment:
this could potentially be a problem for projects that use expo-router but not CNG/dev-client